### ISSUE TITLE
Grammar: KotlinParser.g4:311:0: rule statements contains a closure with at least one alternative that can match EOF

### ISSUE THREAD

**Issue by bennyhuo** [JetBrains/kotlin-spec#103]
I seems to be related to this declaration: 
```
statements
    : (statement (semis statement)*)? semis?
    ;
```

After I cloned the project, I ran ./gradlew :grammar:generateGrammarSource and got this error. Parser cannot be generated.

**Comment by bennyhuo** [comment]
This is caused by a newer version of antlr lib. 

```
antlr - The Antlr libraries to be used for this project.
\--- org.antlr:antlr4:4.+ -> 4.10.1
```

Force the version to 4.4.0 solve the problem.

**Comment by KvanTTT** [comment]
The diagnositcs was introduced to prevent stackoverflow on EOF closures: https://github.com/antlr/antlr4/issues/3359 I'll fix the grammar ASAP.

**Comment by KvanTTT** [comment]
@petukhovv what the meaning of `EOF` inside [`semi`](https://github.com/Kotlin/kotlin-spec/blob/release/grammar/src/main/antlr/KotlinParser.g4#L357) and `semis`? I suggest removing since root rules `kotlinFile` and `script` already have `EOF` at the end. It looks excessive.

**Comment by belyaev-mikhail** [comment]
It is not. Please run grammar tests before suggesting such changes.

**Comment by belyaev-mikhail** [comment]
On the second thought, it may be ok since we don't anymore have any rules that end with semi without ?

All that do contain a raw SEMICOLON token

**Comment by atollk** [comment]
Has this issue been fixed after 13c93b5?
